server/protocols/jmap/src/test/java/org/apache/james/jmap/mailet/ICALBodyTextToAttachmentTest.java

```diff
@@ -0,0 +1,145 @@
+    public void serviceShouldThrowWhenCanNotGetMessageFromMail() throws Exception {
+        expectedException.expect(MailetException.class);
+        Mail mail = mock(Mail.class);
+        when(mail.getMessage())
+            .thenThrow(MessagingException.class);
+
+        mailet.service(mail);
+    }
+
+    @Test
+    public void serviceShouldThrowWhenMessageCanNotGetContentType() throws Exception {
+        expectedException.expect(MailetException.class);
+
+        MimeMessage message = mock(MimeMessage.class);
+        Mail mail = mock(Mail.class);
+        when(message.isMimeType(anyString())).thenThrow(MessagingException.class);
+        when(mail.getMessage()).thenReturn(message);
+
+        mailet.service(mail);
+    }
+
+    @Test
+    public void serviceShouldKeepMessageAsItIsWhenMessageIsNotTextCalendar() throws Exception {
+        String messageContent = "Content-type: text/html; method=REPLY; charset=UTF-8\n" +
+            "Content-transfer-encoding: 8BIT\n" +
+            "\n" +
+            "BEGIN:VCALENDAR";
+        MimeMessage message = MimeMessageBuilder.mimeMessageFromStream(new ByteArrayInputStream(messageContent.getBytes()));
+
+        Mail mail = FakeMail.builder()
+            .mimeMessage(message)
+            .build();
+
+        mailet.service(mail);
+
+        assertThat(mail.getMessage().isMimeType("text/*")).isTrue();
+        assertThat(mail.getMessage().getDisposition()).isNull();
+    }
+
+    @Test
+    public void serviceShouldChangeMessageContentTypeToMultipart() throws Exception {
+        String messageContent = "Content-type: text/calendar; method=REPLY; charset=UTF-8\n" +
+            "Content-transfer-encoding: 8BIT\n" +
+            "\n" +
+            "BEGIN:VCALENDAR";
+        MimeMessage message = MimeMessageBuilder.mimeMessageFromStream(new ByteArrayInputStream(messageContent.getBytes()));
+
+        Mail mail = FakeMail.builder()
+            .mimeMessage(message)
+            .build();
+
+        mailet.service(mail);
+
+        assertThat(mail.getMessage().isMimeType("multipart/*")).isTrue();
+    }
+
+    @Test
+    public void serviceShouldConvertTextBodyOfMessageToAttachmentWhenTextCalendar() throws Exception {
+        String messageContent = "Content-type: text/calendar; method=REPLY; charset=UTF-8\n" +
+            "Content-transfer-encoding: 8BIT\n" +
+            "\n" +
+            "BEGIN:VCALENDAR\n" +
+            "END:VEVENT\n" +
+            "END:VCALENDAR";
+        MimeMessage message = MimeMessageBuilder.mimeMessageFromStream(new ByteArrayInputStream(messageContent.getBytes()));
+
+        Mail mail = FakeMail.builder()
+            .mimeMessage(message)
+            .build();
+
+        mailet.service(mail);
+
+        MimeMessage actual = mail.getMessage();
+        Multipart multipart = (Multipart)actual.getContent();
+
+        assertThat(multipart.getCount()).isEqualTo(1);
+        assertThat(multipart.getBodyPart(0).getDisposition()).isEqualTo("attachment");
+    }
+}
```

Comment:
Can you show me:
 - That you did not alter text/calendar mimepart content?
 - That you did not loose it's headers? Like method, charset, Content-transfer-encoding, etc..

Reply:
>That you did not alter text/calendar mimepart content?

Is that "serviceShouldChangeMessageContentTypeToMultipart"

>That you did not loose it's headers? Like method, charset, Content-transfer-encoding, etc..

I added more UT